src/NzbDrone.Core/MediaFiles/UpdateTrackFileService.cs

```diff
@@ -0,0 +1,140 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NLog;
+using NzbDrone.Common.Disk;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Common.Instrumentation.Extensions;
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.MediaFiles.Events;
+using NzbDrone.Core.Messaging.Events;
+using NzbDrone.Core.Music;
+
+namespace NzbDrone.Core.MediaFiles
+{
+    public interface IUpdateTrackFileService
+    {
+        void ChangeFileDateForFile(TrackFile trackFile, Artist artist, List<Track> tracks);
+    }
+
+    public class UpdateTrackFileService : IUpdateTrackFileService,
+                                            IHandle<ArtistScannedEvent>
+    {
+        private readonly IDiskProvider _diskProvider;
+        private readonly IAlbumService _albumService;
+        private readonly IConfigService _configService;
+        private readonly ITrackService _trackService;
+        private readonly Logger _logger;
+
+        public UpdateTrackFileService(IDiskProvider diskProvider,
+                                        IConfigService configService,
+                                        ITrackService trackService,
+                                        IAlbumService albumService,
+                                        Logger logger)
+        {
+            _diskProvider = diskProvider;
+            _configService = configService;
+            _trackService = trackService;
+            _albumService = albumService;
+            _logger = logger;
+        }
+
+        public void ChangeFileDateForFile(TrackFile trackFile, Artist artist, List<Track> tracks)
```

Comment:
Any reason we have this function and not just the one that it calls?

Reply:
No reason anymore since we only have one redate option unless we think of another date users might want on their files?

Reply:
The only thing I could think of would be actual download date, but that should be done by just not using the redate option.